src/naive/date/serde.rs

```diff
@@ -0,0 +1,78 @@
+use super::NaiveDate;
+use core::fmt;
+use serde::{de, ser};
+
+// TODO not very optimized for space (binary formats would want something better)
+
+impl ser::Serialize for NaiveDate {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        struct FormatWrapped<'a, D: 'a> {
+            inner: &'a D,
+        }
+
+        impl<'a, D: fmt::Debug> fmt::Display for FormatWrapped<'a, D> {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                self.inner.fmt(f)
+            }
+        }
+
+        serializer.collect_str(&FormatWrapped { inner: &self })
+    }
+}
+
+struct NaiveDateVisitor;
+
+impl<'de> de::Visitor<'de> for NaiveDateVisitor {
+    type Value = NaiveDate;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a formatted date string")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        value.parse().map_err(E::custom)
+    }
+}
+
+impl<'de> de::Deserialize<'de> for NaiveDate {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        deserializer.deserialize_str(NaiveDateVisitor)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::naive::date::{test_decodable_json, test_encodable_json};
```

Comment:
I think `test_decodable_json` and `test_encodable_json` should move into `tests`.

Reply:
Is it okay if I do that as a separate PR for these functions in all four modules (including `naive::time`, `naive::datetime` and `datetime`?

Reply:
Yup!